### PR TITLE
Add register name endpoint with basic template

### PR DESCRIPTION
**Pull request by tbille**
# Summary

Add register name endpoint with basic template
Fixes https://github.com/CanonicalLtd/snappy-design-squad/issues/481

# QA

- `./run`
- http://127.0.0.1:8004/account/register-name
- Try to register a snap
- Go on my snaps and check if name is well registered

### CODE REVIEW

**Comment by tbille** [comment]
@Lukewh 

> It would be good to signify that the snap will be registered on the global store.

This a really temporary basic template. I am working on the new template right after: https://raw.githubusercontent.com/CanonicalLtd/snapcraft-design/master/dashboard%20legacy%20work/01%20-%20Register%20snap.png?token=AClQNHoxVsQrYY96EfzqmuEIY8q6aojJks5a-qRwwA%3D%3D

> Can we do 'dry_run' the first time to make sure the name is available too? 

This could be cool, but we need to check how we want to manage this.
It could be nice that when the user goes out of the text input a call would be made to the store to check if the name is available.

File: templates/publisher/register-name.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        </div>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-8">
+        <div class="p-form-validation">
+          <label for="snap-title">Registrant Comment:</label>
+          <div class="p-form-validation__field">
+            <textarea class="p-form-validation__input u-no-margin" name="registrant_comment" rows="10" required></textarea>
```
Comment (bartaz) [inline templates/publisher/register-name.html]: Comment is not required, shouldn't be marked as so.

**Comment by bartaz** [comment]
It would be good to at least show some error messages when registering fails. For example when you try to register already registered or reserved name (like: `firefox`). Currently there is no info at all.

**Comment by Greggless** [comment]
If a name is taken, then they should see something like this:

<hr>

image

<hr>

The 'file a dispute' form is similar to the 'register name' form, but includes a registrant comment and the button is labelled differently

**Comment by tbille** [comment]
@Greggless @Lukewh 
The objective of this PR was mainly to build the backend.
The rest of the template design will be done in this issue: https://github.com/CanonicalLtd/snappy-design-squad/issues/482

**Pull request merged**